src/assignment-cache.ts

```diff
@@ -0,0 +1,65 @@
+import { LRUCache } from 'lru-cache';
+
+export interface AssignmentCacheKey {
+  subjectKey: string;
+  flagKey: string;
+  allocationKey: string;
+}
```

Comment:
the spec for this ticket asked for `subject, allocation and variation` as the caching keys - however i don't think the assigned variation is important for determine whether a subject has logger, that is just a side-effect so we can omit it to save space.

further I thought it was correct to add the `flag`.